src/UI/Art/SpriteArtist/PlayerSetupArtist.java

```diff
@@ -0,0 +1,282 @@
+   */
+  private static class PlayerPanel
+  {
+    // A couple of helper quantities.
+    private static int textVBuffer = 2;
+    private static int portraitPx = SpriteLibrary.getCommanderSprites( "STRONG" ).head.getHeight(); // Faces are square.
+    private static final int EXPECTED_TEXT_LENGTH = 9; // Reasonable expected length for Commander, color, faction, and AI names.
+
+    // Total horizontal panel space, sans scaling.
+    public static final int PANEL_WIDTH = (EXPECTED_TEXT_LENGTH*3 + 2)*SpriteLibrary.getLettersSmallCaps().getFrame(0).getWidth() + textVBuffer*2;
+
+    // Total vertical panel space, sans scaling.
+    public static final int PANEL_HEIGHT = /*Top border*/ 1 + portraitPx + /*name/face border*/ 1
+        + /*portrait buffers*/ 2 + SpriteLibrary.getLettersSmallCaps().getFrame(0).getHeight() + textVBuffer*2 + /*bottom border*/1;
+
+    private String commanderName;
+    private String colorName;
+    private String factionName;
+    private int teamNumber = -99;
+    private String aiName;
+
+    private BufferedImage myImage;
+    private SpriteUIUtils.ImageFrame descriptionPane;
+    private SpriteUIUtils.ImageFrame commanderPane;
+    private SpriteUIUtils.ImageFrame unitPane;
+    private SpriteUIUtils.ImageFrame teamLabel;
+    private SpriteUIUtils.ImageFrame teamPane;
+    private SpriteUIUtils.ImageFrame aiLabel;
+    private SpriteUIUtils.ImageFrame aiPane;
+
+    public PlayerPanel(PlayerSetupInfo info)
+    {
+      myImage = SpriteLibrary.createDefaultBlankSprite( PANEL_WIDTH, PANEL_HEIGHT );
+      Graphics g = myImage.getGraphics();
+      g.setColor(MENUFRAMECOLOR);
+      g.fillRect(0, 0, myImage.getWidth(), myImage.getHeight());
+
+      // Create the two panes that don't ever change.
+      teamLabel = new SpriteUIUtils.ImageFrame(65, 12, 28, 10, MENUHIGHLIGHTCOLOR, MENUBGCOLOR, false, SpriteLibrary.getTextAsImage("TEAM"));
+      aiLabel = new SpriteUIUtils.ImageFrame(94, 12, 54, 10, MENUHIGHLIGHTCOLOR, MENUBGCOLOR, false, SpriteLibrary.getTextAsImage("CONTROL"));
+      teamLabel.render(g);
+      aiLabel.render(g);
+
+      update(info);
+    }
+
+    public BufferedImage update(PlayerSetupInfo info)
+    {
+      // Pull out the info we need.
+      // "Commander-Color Faction"
+      StringBuffer coStrBuf = new StringBuffer(info.getCurrentCO().name);
+      coStrBuf.append("-").append(UIUtils.getPaletteName(info.getCurrentColor())).append(" ").append(info.getCurrentFaction().name);
+
+      boolean cmdrChanged = !info.getCurrentCO().name.equals(commanderName);
+      boolean colorChanged = !UIUtils.getPaletteName(info.getCurrentColor()).equals(colorName);
+      boolean factionChanged = !info.getCurrentFaction().name.equals(factionName);
+      boolean teamChanged = teamNumber != info.getCurrentTeam();
+      boolean aiChanged = !info.getCurrentAI().getName().equals(aiName);
+
+      Graphics g = myImage.getGraphics();
+      if( cmdrChanged || factionChanged )
```

Comment:
When you change color but not faction, the color name does not update (the color does update)

Also, seems odd to define `coStrBuf` up there, and the individual strings down here.

Reply:
Scope narrowed.